src/assets/TokenRatesController.ts

```diff
@@ -54,9 +81,43 @@ export interface TokenRatesConfig extends BaseConfig {
  * Token rates controller state
  *
  * @property contractExchangeRates - Hash of token contract addresses to exchange rates
+ * @property supportedChains - Cached chain data
  */
 export interface TokenRatesState extends BaseState {
-  contractExchangeRates: { [address: string]: number };
+  contractExchangeRates: ContractExchangeRates;
+  supportedChains: SupportedChainsCache;
+}
+
+const COINGECKO_API = {
```

Comment:
[Q] Is this a common pattern in the controller code base, having the UPPER_SNAKE_CASE (which I read as a constant) have methods on it? Non blocking just curious

Reply:
Great question, actually I don't know. It has landed to this point because it started as a constant plain object but then it mutated to an object with methods for convenience. Happy to adopt convention/pattern before moving on.

Reply:
Maybe just `CoinGeckoApi` or some such for now. I could see it living in its own `platforms` file when there are more than one platform we use for this. For now its fine here. Its also fine to let it in like this.

Reply:
Changed on [`1aa1bfb` (#476)](https://github.com/MetaMask/controllers/pull/476/commits/1aa1bfbad2873634dc83ae2ca929f8757e60cc34)